lib/poprouting/src/olsrd_poprouting.c

```diff
@@ -0,0 +1,218 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Visit http://www.olsr.org for more information.
+ *
+ * If you find this software useful feel free to make a donation
+ * to the project. For more information see the website or contact
+ * the copyright holders.
+ *
+ */
+
+#include "olsrd_poprouting.h"
+
+#include <unistd.h>
+
+#include "ipcalc.h"
+#include "builddata.h"
+#include "neighbor_table.h"
+#include "mpr_selector_set.h"
+#include "mid_set.h"
+#include "routing_table.h"
+#include "lq_plugin.h"
+#include "gateway.h"
+#include "gateway_costs.h"
+#include "olsrd_plugin.h"
+#include "info/info_types.h"
+#include "info/http_headers.h"
+#include "gateway_default_handler.h"
+#include "scheduler.h"
+
+float timer=0;
+float hello_mult = 10;
+float tc_mult = 60;
+
+unsigned long long get_supported_commands_mask(void) {
+  return SIW_POPROUTING;
+}
+
+bool isCommand(const char *str, unsigned long long siw) {
+  const char *prefix, s[2] = "=";
+  char *string, *cmd, *s_timer;
+  string = strdup(str);
+  switch (siw) {
+    case SIW_POPROUTING_HELLO:
+      prefix = "/helloTimer";
+      break;
+
+    case SIW_POPROUTING_TC:
+      prefix = "/tcTimer";
+      break;
+
+    case SIW_POPROUTING_HELLO_MULT:
+      prefix = "/helloTimerMult";
+      break;
+
+    case SIW_POPROUTING_TC_MULT:
+      prefix = "/tcTimerMult";
+      break;
+
+    default:
+      return false;
+  }
+  cmd = strtok(string, s);
+  s_timer = strtok(NULL, s);
+  if (s_timer)
+    timer = atof(s_timer);
+  else
+    timer = 0;
+
+  return (!strcmp(cmd, prefix) && timer>=0);
+}
+
+void output_error(struct autobuf *abuf, unsigned int status, const char * req __attribute__((unused)), bool http_headers) {
+  if (http_headers || (status == INFO_HTTP_OK)) {
+    return;
+  }
+
+  /* !http_headers && !INFO_HTTP_OK */
+
+  if (status == INFO_HTTP_NOCONTENT) {
+    /* wget can't handle output of zero length */
+    abuf_puts(abuf, "\n");
+  } else {
+    abuf_appendf(abuf, "error: %s\n", httpStatusToReply(status));
+  }
+}
+
+void set_hello_timer(struct autobuf *abuf) {
+  struct olsr_if *in;
+  assert(abuf);
+  in = olsr_cnf->interfaces;
+  if(!timer){
+    char response[1024];
+    sprintf(response, "hello:%.2f\n", (double)in->cnf->hello_params.emission_interval);
+    abuf_puts(abuf, response);
+    return;
+  }
+  while (in != NULL) {
+    olsr_printf(1, "(POPROUTING) Setting Hello Timer=%f for interface %s\n", (double)timer, in->name);
+    //olsr_change_timer(in->interf->hello_gen_timer, timer * MSEC_PER_SEC, 0, 1);
+    in->interf->hello_gen_timer->timer_period = timer * MSEC_PER_SEC;
+    in->interf->hello_gen_timer->timer_jitter_pct = POPROUTING_JITTER; // Jitter to 5%
```

Comment:
why do this? you overwrite the configured value. I don't like this

Reply:
multiple places

Reply:
i chose to change also the configuration in order to keep the configuration coherent with the timers.